watchdog.go

```diff
@@ -501,3 +505,38 @@ func wdrecover() {
 		}
 	}
 }
+
+// RegisterNotifee registers a function that will be called when a GC has happened.
+// tag is a unique tag that allows unregistering the notifee later using UnregisterNotifee.
+func RegisterNotifee(tag string, f func()) error {
```

Comment:
Could also return a `unregisterFn` here that captures the registered function. That would allow us to do away with the tag, the uniqueness check, and `UnregisterNotifee`. But it comes at the cost of the caller having to save the function in its state.

Nice thing about using a tag is that we can trace the calls -- would you mind adding a `log.Debugf("calling notifee %s" , tag)` when iterating through the map?

Reply:
> Could also return a unregisterFn here that captures the registered function.

I like that! The caller would have to save the tag anyway (and generated a unique tag, which is cumbersome as well, think of a `Service` that can be initialized multiple times, you couldn't use a const string there).